docs/reference/graphql.md

```diff
@@ -0,0 +1,148 @@
+# GraphQL Connector
+
+The [Elastic GraphQL connector](../connectors/sources/graphql.py) is built with the Elastic connectors Python framework and is available as a self-managed [connector client](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+## Availability and prerequisites
+
+This connector is available as a **connector client** from the **Python connectors framework**. To use this connector, satisfy all [connector client requirements](https://www.elastic.co/guide/en/enterprise-search/master/build-connector.html).
+
+This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
+
+## Usage
+
+To use this connector as a **connector client**, select the **GraphQL** tile when creating a new connector under **Search -> Connectors**.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Configuration
+
+### Configure GraphQL connector
+
+Note the following configuration fields:
+
+#### `http_endpoint`  (required)
+
+Provide base URL of the source with endpoint. Example:
+
+ `https://api.xyz.com/graphql`
+
+#### `http_method`  (required)
+
+Select method from GET and POST.
+
+#### `authentication_method`  (required)
+
+Select authentication method from `No Auth`, `Basic Auth`, and `Bearer Token`.
+
+#### `username` (required in case of `Basic Auth`)
+
+In case of `Basic Auth` authentication method, provides username to connect with the source.
+
+#### `password` (required in case of `Basic Auth`)
+
+In case of `Basic Auth` authentication method, provides password to connect with the source.
+
+#### `token` (required in case of `Bearer Token`)
+
+In case of `Bearer token` authentication method, provides token to connect with the source.
+
+#### `graphql_query` (required)
+
+This query will be used to fetch data from the source. The query can contain variables which can be provided in the `graphql_variables` field. The connector will substitute the variables in the query with values from `graphql_variables` and make a GraphQL query to the source. Example:
+
+```
+query getUser($id: ID!) { 
+    user(id: $id) {
+        name 
+        email 
+    }
+}
+```
+
+#### `graphql_variables`
+
+A json of key value pairs of variables used in the GraphQL query. The connector will substitute the variables in the query with the values provided here and make a GraphQL query to the source.
+
+Example: If the GraphQL query is `query getUser($id: ID!) { user(id: $id) { name } }` and the value of `graphql_variables` is `{"id": "123"}`, the final query will be `query getUser { user(id: "123") { name } }` and the connector will make a GraphQL query to the source with the modified query.
+
+#### `graphql_object_list` (required)
```

Comment:
Ahh this variable name slipped through the review, shold we rename it to sth like `graphql_object_id_mapping`

Reply:
As we need to change this throughout the entire connector (graphql.py, test cases, and E2E), would it be okay if I make this change in a follow-up PR?

Reply:
Added the followup ticket https://github.com/elastic/connectors/issues/2420